server/src/org/apache/cloudstack/ca/CAManagerImpl.java

```diff
@@ -0,0 +1,427 @@
+                                    LOG.debug("Failed in auto-renewing certificate for " + hostDescription);
+                                }
+                            } catch (final Throwable ex) {
+                                LOG.warn("Failed to auto-renew certificate for " + hostDescription + ", with error=", ex);
+                                caManager.sendAlert(host, "Certificate auto-renewal failed for " + hostDescription,
+                                        String.format("Certificate is going to expire for %s. Auto-renewal failed to renew the certificate, please renew it manually. It is not valid after %s.", hostDescription, certificate.getNotAfter()));
+                            }
+                        } else {
+                            if (alertMap.containsKey(hostIp)) {
+                                final Date lastSentDate = alertMap.get(hostIp);
+                                if (now.minusDays(1).toDate().before(lastSentDate)) {
+                                    continue;
+                                }
+                            }
+                            caManager.sendAlert(host, "Certificate expiring soon for " + hostDescription,
+                                    String.format("Certificate is going to expire for %s. Please renew it, it is not valid after %s.",
+                                            hostDescription, certificate.getNotAfter()));
+                            alertMap.put(hostIp, new Date());
+                        }
+                    }
+                }
+            } catch (final Throwable t) {
+                LOG.error("Error trying to run CA background task", t);
+            }
+        }
+
+        @Override
+        public Long getDelay() {
+            return CABackgroundJobDelay.value() * 1000L;
+        }
+    }
+
+    public void setCaProviders(final List<CAProvider> caProviders) {
+        this.caProviders = caProviders;
+        initializeCaProviderMap();
+    }
+
+    private void initializeCaProviderMap() {
+        if (caProviderMap != null && caProviderMap.size() != caProviders.size()) {
+            for (final CAProvider caProvider : caProviders) {
+                caProviderMap.put(caProvider.getProviderName().toLowerCase(), caProvider);
+            }
+        }
+    }
+
+    @Override
+    public boolean start() {
+        super.start();
+        initializeCaProviderMap();
+        if (caProviderMap.containsKey(CAProviderPlugin.value())) {
+            configuredCaProvider = caProviderMap.get(CAProviderPlugin.value());
+        }
+        if (configuredCaProvider == null) {
+            LOG.error("Failed to find valid configured CA provider, please check!");
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean configure(final String name, final Map<String, Object> params) throws ConfigurationException {
+        backgroundPollManager.submitTask(new CABackgroundTask(this, hostDao));
```

Comment:
configure is called before start, the bg pool mgr's executor is started at start() which uses a dynamic size based on the submitted tasks list/collection so things need to be submitted/configured in `configure()` of any manager/service impl. See, the background polling manager implementation you'll like what I've done there :)